tests/rados/scheduled_scrub_scenarios.py

```diff
@@ -0,0 +1,82 @@
+"""
+   This module contains the methods required for scrubbing.
+   1.To set the parameters for scrubbing initially required the
+     cluster time and day details.get_cluster_date method provides
+     the details.
+   2.set_osd_configuration method  used to set the configuration
+     parameters on the cluster.
+   3.get_osd_configuration  method is used to get the configured parameters
+     on the cluster.
+     NOTE: With set_osd_configuration & get_osd_configuration methods can
+          use to set the get the any OSD configuration parameters.
+   4. get_pg_dump  method is used to get the pg dump details from the cluster.
+   5. verify_scrub  method used for the verification of scheduled scrub.
+"""
+
+import logging
+import time
+from datetime import datetime
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.rados_scrub import RadosScrubber
+
+log = logging.getLogger(__name__)
+
+
+def run(ceph_cluster, **kw):
+
+    OSD_SCRUB_MIN_INTERVAL = 1800
+    OSD_SCRUB_MAX_INTERVAL = 3600
+    OSD_DEEP_SCRUB_INTERVAL = 3600
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosScrubber(node=cephadm)
+
+    # Storing the pg dump log before setting the scrub parameters
+    before_scrub_log = rados_obj.get_pg_dump("pgid", "last_scrub_stamp")
+
+    # Get the cluster time
+    current_time = rados_obj.get_cluster_date()
+
+    # Preparation of cofiguration parameter values from the current
+    # cluster time
+    yy, mm, dd, hh, day = current_time.split(":")
+    date = datetime.datetime(int(yy), int(mm), int(dd), int(hh))
+    date += datetime.timedelta(hours=1)
+    scrub_begin_hour = date.hour
+    date += datetime.timedelta(hours=1)
+    scrub_end_hour = date.hour
+    date += datetime.timedelta(days=1)
+    scrub_begin_weekday = date.weekday()
+    if scrub_end_hour == 0:
+        date += datetime.timedelta(days=1)
+        scrub_end_weekday = date.weekday()
+    else:
+        scrub_end_weekday = date.weekday()
+
+    # Setting the scrub parameters
+    rados_obj.set_osd_configuration("osd_scrub_min_interval", OSD_SCRUB_MIN_INTERVAL)
+    rados_obj.set_osd_configuration("osd_scrub_max_interval", OSD_SCRUB_MAX_INTERVAL)
+    rados_obj.set_osd_configuration("osd_deep_scrub_interval", OSD_DEEP_SCRUB_INTERVAL)
+    rados_obj.set_osd_configuration("osd_scrub_begin_week_day", scrub_begin_weekday)
+    rados_obj.set_osd_configuration("osd_scrub_end_week_day", scrub_end_weekday)
+    rados_obj.set_osd_configuration("osd_scrub_begin_hour", scrub_begin_hour)
+    rados_obj.set_osd_configuration("osd_scrub_end_hour", scrub_end_hour)
+
+    # Scheduled scrub verification
+    endTime = datetime.datetime.now() + datetime.timedelta(minutes=60)
+    while True:
+        if datetime.datetime.now() >= endTime:
+            break
```

Comment:
you could do 
```
while datetime.datetime.now() > end_time
```
this would avoid the extra `if` statement

Reply:
Fixed the issue.